### ISSUE TITLE
Remove upper bound on python_requires

### ISSUE THREAD

**Issue by jmahlik** [aws/aws-sdk-pandas#1714]
### Describe the bug

Propose removing the upper bound on python requires to allow users to test with python 3.11 in their projects. The python requires upper bound ideally shouldn't be set unless there's a very specific reason to do so, this just happened to be the default poetry chose at some point, so many project seem to have adopted it.

Some interesting articles on why this might not be ideal:

* https://iscinumpy.dev/post/bound-version-constraints/#Pinning%20the%20Python%20version%20is%20special
* https://discuss.python.org/t/requires-python-upper-limits/12663

```patch
diff --git a/pyproject.toml b/pyproject.toml
index a36f685..3be566d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -25,7 +25,7 @@ classifiers = [
 ]

 [tool.poetry.dependencies]
-python = ">=3.7.1, <3.11"
+python = ">=3.7.1"

 boto3 = "^1.24.11"
 botocore = "^1.27.11"
```

p.s. I have attempted install on python 3.11 with the edited pyproject, some of the deps don't have wheels yet, so it might not be possible to test this in ci. That's seems fine though, just add the classifier once it is tested. At least users will be able to upgrade at their own pace without wrangler being a hard block for an odd choice by poetry once the wheels are released.

### How to Reproduce

Include awswrangler as a dep and use python 3.11.

### Expected behavior

To be able to run the test suites of our ~10 projects including awswrangler on python 3.11 to see what c extension dependencies may need wheels released still. Right now, wrangler causes a hard fail for not such a great reason IMO.

### OS

windows/linux

### Python version

3.11

### AWS SDK for pandas version

installed from main

**Comment by malachi-constant** [comment]
@jmahlik Appreciate you starting the discussion. I personally, agree with removing the `py` version limitation as long as we make clear in our [docs](https://aws-sdk-pandas.readthedocs.io/en/stable/install.html), which versions have been thoroughly tested compatibility. Thoughts from the rest of the team?

**Comment by jaidisido** [comment]
Thanks @jmahlik, we thoroughly test the library against every python version we support as we have noticed differences in behaviour in the past. However, I agree that this should not be a blocker for users wishing to install the package on the latest version. So in favour of removing the limit as well

**Comment by jmahlik** [comment]
Any timeframe for the next release from main? Asking because it'd be nice to have this incorporated in the 2.0 series before 3.0 gets released so we don't have to bump to wrangler 3.0 right away to get 3.11 support.

Side note for anyone installing on windows, looks like there's two deps that haven't shipped windows wheels just yet.

* https://github.com/lxml/lxml/pull/360
* https://github.com/aio-libs/aiohttp/issues/6600

**Comment by kukushking** [comment]
@jmahlik working on 2.18.0 - expected today

**Comment by kjyv** [comment]
Actually we can't install awswrangler on python 3.11.1.
It seems you're still depending on aiohttp 3.8.1 which does not have python 3.11 support. It should be at least 3.8.2 but better 3.8.3 as it's supposed to have more fixes for 3.11.

**Comment by phillipuniverse** [comment]
Ugh 3.6.2 was released this morning and did not have the aiohttp change. I asked about it [on their mailing list](https://groups.google.com/g/gremlin-users) so we'll see what they say.

**Comment by phillipuniverse** [comment]
I opened a PR to backport the aiohttp change to 3.6 based on the mailing list discussion at https://groups.google.com/g/gremlin-users/c/nZ-mSV2iMlo/m/i4k7JJAcAwAJ

**Comment by jmahlik** [comment]
I do think it might be worth discussing the dependency situation in another issue. There's a lot of deps in here that one might consider "extras". ex. One wants to use s3, but one needs to install ~15 transitive dependencies for the neptune functionality.

**Comment by jmahlik** [comment]
> I do think it might be worth discussing the dependency situation in another issue. There's a lot of deps in here that one might consider "extras". ex. One wants to use s3, but one needs to install ~15 transitive dependencies for the neptune functionality.

https://github.com/aws/aws-sdk-pandas/issues/1983

**Comment by phillipuniverse** [comment]
@jmahlik @kukushking @kjyv FYI the conclusion of https://github.com/apache/tinkerpop/pull/1958 is that the relaxed version constraint for aiohttp required for Python 3.11 is now backported to both the 3.5 and 3.6 lines of gremlinpython.

They also released versions 3.6.3rc1/3.5.6rc1 that includes this dependency version change.

I tested this by changing my `pyproject.toml`, this is now necessary for Python 3.11 support for awswrangler, might be worth documenting somewhere?

```toml
awswrangler = "^2.19"
gremlinpython = "3.6.3rc1"
```

It's also likely this is rendered moot by the changes at #1992
